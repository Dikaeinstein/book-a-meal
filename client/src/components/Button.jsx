import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
