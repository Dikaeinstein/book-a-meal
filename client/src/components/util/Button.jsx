import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, ...otherProps }) => (
  <button {...otherProps}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
