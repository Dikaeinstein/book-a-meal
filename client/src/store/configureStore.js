import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
