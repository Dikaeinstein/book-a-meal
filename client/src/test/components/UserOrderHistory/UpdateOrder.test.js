import React from 'react';
import { shallow } from 'enzyme';
import sweetalert from 'sweetalert';
import configureStore from 'redux-mock-store';
import ConnectedUpdateOrder, { UpdateOrder } from
  '../../../components/UserOrderHistory/UpdateOrder';
import Button from '../../../components/util/Button';


jest.mock('sweetalert');

describe('<UpdateOrder />', () => {
  const order = {
    id: 11,
    quantity: 1,
    total: '4000',
    status: 'delivered',
    expired: false,
    mealId: 3,
    userId: 4,
    meal: {
      id: 3,
      name: 'Amala and Ewedu',
      price: '4000',
    },
    user: {
      id: 4,
      name: 'Chinonso Okorie',
    },
  };
  let wrapper;
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({ orders: { isUpdating: false } });
    wrapper = shallow(<ConnectedUpdateOrder
      store={store}
      defaultOrder={order}
      closeModal={jest.fn()}
    />);
    expect(wrapper.props().isUpdating).toBeFalsy();
  });
  it('should call handleChange', () => {
    const spy = jest.spyOn(UpdateOrder.prototype, 'handleChange');
    wrapper = shallow(<UpdateOrder
      defaultOrder={order}
      closeModal={jest.fn()}
      modifyOrder={jest.fn()}
    />);
    const quantity = wrapper.find('#quantity');
    quantity.simulate('change', {
      preventDefault: jest.fn(),
      target: { value: 2 },
    });
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().quantity).toBe(2);
  });
  it('should call handleQuantityUpdate', () => {
    const spy = jest.spyOn(UpdateOrder.prototype, 'handleQuantityUpdate');
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper = shallow(<UpdateOrder
      defaultOrder={order}
      closeModal={jest.fn()}
      modifyOrder={jest.fn()}
    />);
    wrapper.find(Button).first().simulate('click');
    sweetalert.mockResolvedValue(Promise.reject());
    wrapper.find(Button).first().simulate('click');
    sweetalert.mockResolvedValue(Promise.reject(new Error()));
    wrapper.find(Button).first().simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    const tree = shallow(<UpdateOrder
      defaultOrder={order}
      closeModal={jest.fn()}
      modifyOrder={jest.fn()}
    />);
    expect(tree).toMatchSnapshot();
  });
});
