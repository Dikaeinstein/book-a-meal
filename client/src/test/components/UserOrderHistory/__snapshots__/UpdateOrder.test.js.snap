// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateOrder /> should match snapshot 1`] = `
<div
  className="text-center"
  id="checkout"
>
  <h5>
    Update Order
  </h5>
  <div>
    <table>
      <thead
        className="text-center"
      >
        <tr>
          <th>
            Meal Name
          </th>
          <th>
            Quantity (Plate)
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="text-center"
        >
          <td>
            Amala and Ewedu
          </td>
          <td>
            <input
              id="quantity"
              max={9007199254740991}
              min={1}
              name="quantity"
              onChange={[Function]}
              style={
                Object {
                  "width": "50%",
                }
              }
              type="number"
              value={1}
            />
          </td>
          <td>
            â‚¦ 
            4000
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginTop": "1rem",
      }
    }
  >
    <Button
      className="btn btn-default"
      onClick={[Function]}
      style={
        Object {
          "background": "#28a745",
          "margin": "1rem .5rem",
        }
      }
      title="update order"
      value="Update Order"
    />
    <Button
      className="btn btn-default"
      onClick={[MockFunction]}
      style={
        Object {
          "background": "#dc3545",
          "margin": "1rem .5rem",
        }
      }
      value="Cancel"
    />
  </div>
</div>
`;
